Modules/Core/Transform/include/itkResampleInPlaceImageFilter.h

```diff
@@ -0,0 +1,160 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkResampleInPlaceImageFilter_h
+#define itkResampleInPlaceImageFilter_h
+
+#include "itkImageToImageFilter.h"
+#include "itkVersorRigid3DTransform.h"
+
+namespace itk
+{
+/** \class ResampleInPlaceImageFilter
+ * \brief Resample an image in place.
+ *
+ * The current ITK resample image filter will generate a physical memory-
+ * modified version of the input image if the input transform is not identity. The
+ * abuse use of the filter can be cumbersome in a situation that the image is very
+ * large, and there are lots of transform to be superimposed for the input image, and
+ * we even don't care about those intermediate transformed images.
+ *
+ * If all the transforms are rigid, a far superior way to achieve a similar result
```

Comment:
Not sure if I understand why transformations are mentioned in the preceding paragraph and here if the filter name and brief state that this "only" resamples an image in place.

Reply:
I reworded most of the documentation now.